src/SAWScript/Crucible/MIR/Override.hs

```diff
@@ -117,6 +136,400 @@ assignTerm sc cc md prepost var val =
        Just old ->
          matchTerm sc cc md prepost val old
 
+-- | When a specification is used as a composition override, this function
+-- checks that the postconditions of the specification fully specify (via
+-- @mir_points_to@ statements) the values of all local mutable allocations
+-- (which are declared in the preconditions via @mir_alloc_mut@) and all
+-- mutable static items. If not, this function will raise an appropriate error
+-- message. See @Note [MIR compositional verification and mutable allocations]@.
+checkMutableAllocPostconds ::
+  Options ->
+  SharedContext ->
+  MIRCrucibleContext ->
+  CrucibleMethodSpecIR ->
+  OverrideMatcher MIR md ()
```

Comment:
A key function that checks if compositional MIR overrides are well-formed with respect to mutable allocations. This is the part of the implementation that is the most interesting, as it isn't simply cargo-culted from the LLVM backend.